### PR TITLE
Clear only once per connection

### LINKED ISSUE #698: Graph data sent to runtime multiple times when refreshing browser
To reproduce

* Have a project open in Flowhub, with a runtime connected
* Refresh the page (using Ctrl+F5 or similar)
* Observe communication going to runtime on WebSocket. Either on runtime side or using Chrome devtools

Expected:
`graph:clear`, `graph:addnode` and similar commands are sent *once*

Actual:
`graph:clear`, `graph:addnode`, ... is sent *three* times

### PR DESCRIPTION
**Pull request by bergie**
Will fix #698

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "noflo-objects": "~0.2.0",
     "noflo-polymer": "~0.2.0",
     "noflo-routers": "~0.2.2",
-    "noflo-runtime": "~0.3.13",
+    "noflo-runtime": "~0.3.14",
```
Comment (jonnor) [inline package.json]: Can we move the components from noflo-runtime up here soon? They seem very tightly connected...

File: components/ContextToRuntime.coffee [hunk 2]
```diff
@@ -104,21 +112,17 @@ exports.getComponent = ->
           # Different runtime, different graph. Disconnect old runtime connection
           currentContext.runtime.disconnect()
 
+      send = _.debounce sendContext, 300, true
```
Comment (jonnor) [inline components/ContextToRuntime.coffee]: Ugh, why?

File: components/ContextToRuntime.coffee [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
     return
 
   if context.graphs?.length
     sendGraph graph, context.runtime for graph in context.graphs
+    out.send context
+    out.disconnect()
     return
 
+  out.send context
+  out.disconnect()
```
Comment (jonnor) [inline components/ContextToRuntime.coffee]: Can't we use `sendDone()` or whatsitcalled?

Comment (bergie) [inline components/ContextToRuntime.coffee]: We're not on 0.8 yet

File: components/ContextToRuntime.coffee (hunk 2 above)
Comment (bergie) [inline components/ContextToRuntime.coffee]: So we don't send more than once in this situation where we happen to get the runtime just between when it had connected, but before it had received capabilities

File: package.json (hunk 1 above)
Comment (bergie) [inline package.json]: Yes, we should do that sooner or later

File: components/ContextToRuntime.coffee (hunk 2 above)
Comment (jonnor) [inline components/ContextToRuntime.coffee]: Buffering until we receive capabilities would seem less racy than a time-based bump...

File: components/ContextToRuntime.coffee (hunk 3 above)
Comment (jonnor) [inline components/ContextToRuntime.coffee]: Ok

**Pull request merged**